### ISSUE TITLE
Collaboration is broken.

### ISSUE THREAD

**Issue by Aniket21mathur** [sugarlabs/readetexts#15]
Tested with Ubuntu 18.04 Vm. Sugar v 0.112.
Logs
```
Traceback (most recent call last):
  File "/home/guest4/Activities/ReadETexts.activity/ReadEtextsActivity.py", line 1415, in download_document
    utf8_strings=True)
  File "/usr/lib/python2.7/dist-packages/dbus/proxies.py", line 145, in __call__
    **keywords)
  File "/usr/lib/python2.7/dist-packages/dbus/connection.py", line 651, in call_blocking
    message, timeout)
dbus.exceptions.DBusException: org.freedesktop.Telepathy.Error.InvalidArgument: Tube is not in the local pending state
```
I am testing using a Vm on localhost using rdesktop. Does it have any effect.
@quozl @chimosky please have a look. Thanks!

**Comment by quozl** [comment]
Code looks wrong, note the reference to `shared_activity`.  My guess is that collaboration wasn't properly ported in fd972f1ca3da5e0f5b418709f2b88e9ae422fc9f, and will have to be completed first.

**Comment by Aniket21mathur** [comment]
> Code looks wrong, note the reference to `shared_activity`

Thanks, I don't think use of shared_activity is causing this issue. In fact use of shared_activity do not produce any error on sugar v 0.112 Ubuntu 18.04. I verified it by changing all shared_activity instances to get_shared_activity(), but still the above traceback remains. I fixed it in #16 , please see.
